angr_platforms/risc_v/simos_riscv.py

```diff
@@ -0,0 +1,56 @@
+from angr.simos import SimOS, register_simos
+from angr.sim_procedure import SimProcedure
+from angr.engines.vex import SimEngineVEX
+from angr.calling_conventions import SimStackArg, SimRegArg, SimCC, register_syscall_cc, register_default_cc, SimCC
+from .arch_riscv import ArchRISCV
+
+
+class SimCCRISCV(SimCC):
+    ARG_REGS = [ 'a0', 'a1', 'a2', 'a3', 'a4', 'a5','a6', 'a7']
+    FP_ARG_REGS = []    # TODO: ???
+    STACK_ALIGNMENT = 16
+    RETURN_ADDR = SimStackArg(4, 4)
+    RETURN_VAL = SimRegArg('ra', 4)
+    ARCH = ArchRISCV
+
+class SimRISCV(SimOS):
+    # Syscalls are for lamers
+    SYSCALL_TABLE = {}
+
+
+    def __init__(self, *args, **kwargs):
+        super(SimRISCV, self).__init__(*args, name='RISCV', **kwargs)
+
+    def configure_project(self):
+        super(SimRISCV, self).configure_project()
+
+        #self._load_syscalls(SimMSP430.SYSCALL_TABLE, "bf")
+
+    def state_blank(self, data_region_size=0x8000, **kwargs):
+        # pylint:disable=arguments-differ
+        state = super(SimRISCV, self).state_blank(**kwargs)  # pylint:disable=invalid-name
+        # PTR starts halfway through memory
+        return state
```

Comment:
Is there something missing here?

Reply:
As you can see from a couple of lines before I used the MSP430 platform as base and modified it to the RISC-V. The PTR comment is a relic form the MSP430 code and can be removed

Reply:
Right, I can see that, but you've basically removed every single thing that makes this class different from the base SimOS. I'm pretty sure the whole class can be removed.

Reply:
Removing the entire class would lead to the loss of the name='RISCV' argument set in the init function. Wouldn't that cause any issues? I do agree on removing the configure_project, state_blank and state_entry functions.

Reply:
Considering that risc-v is not the name of any OS I know of, I think we're fine. There are no functional uses of the name field, so far as I can tell.

Reply:
I removed the default SimOS and added a testcase as suggested